3478/2/utils/photo_downloader.rb

```diff
@@ -0,0 +1,13 @@
+require 'open-uri'
+require 'json'
+
+module PhotoDownloader
+  API_TG_LINK = 'https://api.telegram.org/bot'.freeze
+  GET_ID_LINK = '/getFile?file_id='.freeze
+
+  def download_photo
+    link = API_TG_LINK + ENV['TELEGRAM_TOKEN'] + GET_ID_LINK + session[:photo_id]
+    file_path = JSON.parse(URI.open(link).read)['result']['file_path']
+    URI.open("https://api.telegram.org/file/bot#{ENV['TELEGRAM_TOKEN']}/#{file_path}").read
```

Comment:
You have constant?

Reply:
Obviously. Is it wrong in any way. I'm not certain what should I change about it

Reply:
`API_TG_LINK` ` GET_ID_LINK` `"https://api.telegram.org/file/bot"` I think now you understand. But yea you can solve it much more easy